app/src/main/java/edu/nextstep/camp/calculator/MainPresenter.kt

```diff
@@ -26,7 +26,7 @@ class MainPresenter(
 
     override fun proceedCalculation() {
         val result = calculator.calculate(expression.toString())
-        if (result == null) {
+        if (result == null || expression.toString() == result.toString()) {
             view.showIncompleteExpression()
             return
         }
```

Comment:
OOP의 핵심중 하나는 객체간의 '협력'으로 협력을 하려면 잘 '물어봐야' 합니다. 
https://m.blog.naver.com/PostView.naver?isHttpsRedirect=true&blogId=sorang226&logNo=221796702354
(원문 https://martinfowler.com/bliki/TellDontAsk.html)

```suggestion
        if (result == null || expression.isNotChanged(result)) {
            view.showIncompleteExpression()
            return
        }
```

도메인 객체에서 무언가를 가져와 비교하기보다는 도메인 객체에 물어보는 형태로 작성해보세요! 🙏
그런데 버그는 여전히 존재합니다. ㅠ
완벽히 버그 없이 구현되었냐를 보는건 아니어서 버그자체는 넘어가도록 하겠습니다!